packages/nuxt/src/pages/module.ts

```diff
@@ -266,36 +263,14 @@ export default defineNuxtModule({
       })
     })
 
-    nuxt.hook('nitro:init', (nitro) => {
-      if (nuxt.options.dev || !nitro.options.static || nuxt.options.router.options.hashMode) { return }
-      // Prerender all non-dynamic page routes when generating app
-      const prerenderRoutes = new Set<string>()
-      nuxt.hook('pages:extend', (pages) => {
-        prerenderRoutes.clear()
-        const processPages = (pages: NuxtPage[], currentPath = '/') => {
-          for (const page of pages) {
-            // Add root of optional dynamic paths and catchalls
-            if (OPTIONAL_PARAM_RE.test(page.path) && !page.children?.length) { prerenderRoutes.add(currentPath) }
-            // Skip dynamic paths
-            if (page.path.includes(':')) { continue }
-            const route = joinURL(currentPath, page.path)
-            prerenderRoutes.add(route)
-            if (page.children) { processPages(page.children, route) }
-          }
-        }
-        processPages(pages)
-      })
-      nuxt.hook('nitro:build:before', (nitro) => {
-        if (nitro.options.prerender.routes.length) {
-          for (const route of nitro.options.prerender.routes) {
-            // Skip default route value as we only generate it if it is already
-            // in the detected routes from `~/pages`.
-            if (route === '/') { continue }
-            prerenderRoutes.add(route)
-          }
-        }
-        nitro.options.prerender.routes = Array.from(prerenderRoutes)
-      })
+    nuxt.hook('app:resolve', (app) => {
+      const nitro = useNitro()
+      if (nitro.options.prerender.crawlLinks) {
```

Comment:
This condition plus nitro not scanning `/` by default means the pages directory routes are no longer prerendered automatically. Was this intentional?